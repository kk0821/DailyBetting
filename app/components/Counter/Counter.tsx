import * as React from 'react';
import { Container, Button, Text } from 'native-base';

export interface CounterProps {
  count: number;
  onDecrement?: () => any;
  onIncrement?: () => any;
}

export const Counter: React.FunctionComponent<CounterProps> = (
  props: CounterProps
) => (
  <Container>
    <Button small={true} onPress={props.onDecrement}>
      <Text>-</Text>
    </Button>
    <Text>{props.count}</Text>
    <Button small={true} onPress={props.onIncrement}>
      <Text>+</Text>
    </Button>
  </Container>
);

Counter.defaultProps = {
  onDecrement: () => {},
  onIncrement: () => {}
};
