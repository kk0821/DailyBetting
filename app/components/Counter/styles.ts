import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  Counter: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row'
  }
});
