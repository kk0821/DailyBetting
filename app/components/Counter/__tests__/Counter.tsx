import * as React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'native-base';
import { Counter } from '../Counter';

it('renders correctly with defaults', () => {
  const counter = shallow(<Counter count={10} />);
  expect(counter).toMatchSnapshot();
});

it('calls the onIncrement handler when the + button is pressed', () => {
  const onIncrement = jest.fn();
  const counter = shallow(<Counter count={10} onIncrement={onIncrement} />);
  counter
    .find(Button)
    .at(1)
    .simulate('press');
  expect(onIncrement).toHaveBeenCalledTimes(1);
});

it('calls the onDecrement handler when the - button is pressed', () => {
  const onDecrement = jest.fn();
  const counter = shallow(<Counter count={10} onDecrement={onDecrement} />);
  counter
    .find(Button)
    .at(0)
    .simulate('press');
  expect(onDecrement).toHaveBeenCalledTimes(1);
});
