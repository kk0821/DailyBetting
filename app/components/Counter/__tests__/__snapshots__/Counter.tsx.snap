// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<Styled(Container)>
  <Styled(Button)
    onPress={[Function]}
    small={true}
  >
    <Styled(Text)>
      -
    </Styled(Text)>
  </Styled(Button)>
  <Styled(Text)>
    10
  </Styled(Text)>
  <Styled(Button)
    onPress={[Function]}
    small={true}
  >
    <Styled(Text)>
      +
    </Styled(Text)>
  </Styled(Button)>
</Styled(Container)>
`;
